src/main/scala/mesosphere/marathon/core/volume/claims/VolumeClaimsDelegate.scala

```diff
@@ -0,0 +1,139 @@
+package mesosphere.marathon.core.volume.claims
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Props }
+import akka.event.LoggingReceive
+import akka.util.Timeout
+import mesosphere.marathon.core.volume._
+import mesosphere.marathon.state.{ PathId, PersistentVolume, Volume }
+import org.slf4j.LoggerFactory
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.concurrent.duration._
+
+import akka.pattern.ask
+
+private[claims] object VolumeClaimsDelegate {
+  import VolumeClaims._
+
+  case class CreateClaimRequest(c: Claim, l: Listener)
+  case class DestroyClaimRequest(t: ClaimToken)
+}
+
+protected[volume] class VolumeClaimsDelegate(actorRef: ActorRef) extends VolumeClaims {
+  import VolumeClaims._
+  private[this] implicit val timeout: Timeout = 1.second
+
+  override def create(c: Claim, l: Listener): Unit = {
+    actorRef ! VolumeClaimsDelegate.CreateClaimRequest(c, l)
+  }
+  override def destroy(t: ClaimToken): Unit = {
+    actorRef ! VolumeClaimsDelegate.DestroyClaimRequest(t)
+  }
+}
+
+protected[volume] object VolumeClaimsActor {
+  import VolumeClaims._
+
+  private val log = LoggerFactory.getLogger(getClass)
+  // TODO(jdef) move to Claim
+  private val emptyClaim = ClaimEntry(Claim("", PathId.empty, Set.empty[Volume]), None, None)
+
+  case class ClaimEntry(claim: Claim, token: Option[ClaimToken], listener: Option[Listener]) {
+    def withToken(t: ClaimToken): ClaimEntry = ClaimEntry(claim, Some(t), listener)
+  }
+
+  case class VolumeName(name: String, providerName: String)
+  object VolumeName {
+    def apply(v: Volume): Option[VolumeName] = {
+      Some(v).collect{
+        case pv: PersistentVolume =>
+          VolumeName(pv.persistent.name.getOrElse(""), pv.persistent.providerName.getOrElse(""))
+      }
+    }
+  }
+
+  def props: Props = Props(new VolumeClaimsActor)
+}
+
+private[volume] class VolumeClaimsActor extends Actor with ActorLogging {
+  import VolumeClaims._
+  import VolumeClaimsActor._
+  import VolumeClaimsDelegate._
+
+  // all claims (pending + bound)
+  private[this] var claimsById: Map[String, ClaimEntry] = Map.empty
+
+  // bound claims
+  private[this] var claimsByToken: Map[ClaimToken, ClaimEntry] = Map.empty
+
+  override def receive: Receive = LoggingReceive {
+    Seq[Receive](
+      receiveClaimRequest
+    ).reduceLeft(_.orElse[Any, Unit](_))
+  }
+
+  private[this] def receiveClaimRequest: Receive = {
+    // TODO(jdef) consider an update to an existing claim (app config changed..)
```

Comment:
nothing to consider along these lines here. AppTaskLauncherActor is responsible for sync'ing claims according to app config changes. this actor just processes requests that are generated from that actor.